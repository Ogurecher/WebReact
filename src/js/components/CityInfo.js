import React, { Component } from "react";
import CityLabel from "./CityLabel.js";
import WeatherIcon from "./WeatherIcon.js";
import WeatherList from "./WeatherList.js";
/*import { getWeather } from "../actions/index";
import { connect } from "react-redux";*/
import { LOADING_GIF_URL } from "../constants/resources.js"
import CloseButton from "./CloseButton.js";

export default class CityInfo extends Component {

    constructor(props) {
        super(props);

        this.renderError = this.renderError.bind(this);
        this.renderInfo = this.renderInfo.bind(this);
        this.renderLoading = this.renderLoading.bind(this);
    }

    /*componentDidUpdate(prevProps) {
        if (JSON.stringify(this.props.position) !== JSON.stringify(prevProps.position)) {
            this.props.getWeather(this.props.position);
        }
    }*/
    
    renderError(msg) {
        return (
            <div>
                <p> {msg} </p>
                <CloseButton city={this.props.city} />
            </div>
        );
    }

    renderInfo() {
        return (
            <div>
                <CityLabel weather={this.props.city.weather} />
                <WeatherIcon weather={this.props.city.weather} />
                <CloseButton city={this.props.city} />
                <WeatherList weather={this.props.city.weather} />
            </div>
        );
    }

    renderLoading() {
        return (
            <div>
                <p> LOADING... </p>
                <img src={LOADING_GIF_URL} />
            </div>
        );
    }

    render() {
        if (this.props.city.errorMsg) {
            return this.renderError(this.props.city.errorMsg);
        } else if (this.props.city.weather && this.props.city.loading === 0) {
            return this.renderInfo();
        } else if (this.props.city.loading === 1) {
            return this.renderLoading();
        } else {
            return(<p>No CityInfo</p>);                 //REMOVE DEBUG MSG
        }
    }
}