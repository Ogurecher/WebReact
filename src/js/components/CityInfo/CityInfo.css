.city-info {
    display: flex;
    flex-direction: column;
    flex: 0 0 40%;
    margin-left: 5%;
    margin-right: 5%;
}

.city-info-head {
    display: flex;
    flex-direction: row;
    align-items: center;
    justify-content: space-between;
    font-size: 150%;
    padding: 1%;
}

.city-info-loading {
    display: flex;
    flex-direction: column;
    align-items: center;
    justify-content: center;
    flex: 0 0 40%;
    margin-left: 5%;
    margin-right: 5%;
    height: 400px;
}

.city-info-error {
    display: flex;
    flex-direction: column;
    align-items: center;
    justify-content: center;
    flex: 0 0 40%;
    margin-left: 5%;
    margin-right: 5%;
    height: 400px;
}