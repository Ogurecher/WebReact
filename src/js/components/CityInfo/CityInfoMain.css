#main {
    flex-direction: row;
    flex: 0 0 100%;
}

#main > div:first-child {
    font-size: 300%;
    flex-wrap: wrap;
    align-content: flex-start;
    padding: 2%;
}

#main > div:first-child > p:first-child {
    flex: 0 1 100%;
}

#main > div > form > input {
    visibility: hidden;
}