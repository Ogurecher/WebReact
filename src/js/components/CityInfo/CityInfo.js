import React, { Component } from 'react';
import CityLabel from '../CityLabel/CityLabel';
import WeatherIcon from '../WeatherIcon/WeatherIcon';
import WeatherList from '../WeatherList/WeatherList';
import { LOADING_GIF_URL } from '../../constants/resources.js'
import CloseButton from '../CloseButton/CloseButton';
import Temperature from '../Temperature/Temperature';
import './CityInfo.css';
import './CityInfoMain.css';

export default class CityInfo extends Component {

    constructor(props) {
        super(props);

        this.renderError = this.renderError.bind(this);
        this.renderInfo = this.renderInfo.bind(this);
        this.renderLoading = this.renderLoading.bind(this);
    }
    
    renderError(msg) {
        return (
            <div className='city-info-error'>
                <p> {msg} </p>
                    <CloseButton city={this.props.city} />
            </div>
        );
    }

    renderInfo() {
        return (
            <div id={this.props.id} className='city-info'>
                <div className='city-info-head'>
                    <CityLabel weather={this.props.city.weather} />
                    <WeatherIcon weather={this.props.city.weather} />
                    <Temperature weather={this.props.city.weather} />
                    <CloseButton city={this.props.city} />
                </div>
                <WeatherList weather={this.props.city.weather} />
            </div>
        );
    }

    renderLoading() {
        return (
            <div className='city-info-loading'>
                <p> Loading, please wait... </p>
                <img src={LOADING_GIF_URL} />
            </div>
        );
    }

    render() {
        if (this.props.city.errorMsg) {
            return this.renderError(this.props.city.errorMsg);
        } else if (this.props.city.weather && this.props.city.loading === 0) {
            return this.renderInfo();
        } else if (this.props.city.loading === 1) {
            return this.renderLoading();
        } else {
            return(<p></p>);
        }
    }
}