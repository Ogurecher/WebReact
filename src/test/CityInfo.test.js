import React from 'react';
import renderer from 'react-test-renderer';
import configureMockStore from 'redux-mock-store';
import { Provider } from 'react-redux';

import CityInfo from '../js/components/CityInfo/CityInfo';

const store = configureMockStore();

describe('CityInfo', function() {
    test('snapshot renders info', function() {
        const weatherStub = {
            name: 'London',
            wind: {
                deg: 'winddeg',
                speed: 'windspd'
            },
            weather: [{
                description: 'desc',
                icon: '50n'
            }],
            main: {
                pressure: 'pressure',
                humidity: 'humidity'
            },
            coord: {
                lat: 'lat',
                lon: 'lon'
            }
        };

        const cityStub = {
            weather: weatherStub,
            loading: 0,
            errorMsg: null
        };

        const storeMock = store();

        const component = renderer.create(<Provider store={storeMock}><CityInfo city={cityStub} /></Provider>);
        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });

    test('snapshot renders loading', function() {
        const cityStub = {
            loading: 1
        };

        const storeMock = store();

        const component = renderer.create(<Provider store={storeMock}><CityInfo city={cityStub} /></Provider>);
        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });

    test('snapshot renders error', function() {
        const cityStub = {
            errorMsg: 'Location not found'
        };

        const storeMock = store();

        const component = renderer.create(<Provider store={storeMock}><CityInfo city={cityStub} /></Provider>);
        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });
});