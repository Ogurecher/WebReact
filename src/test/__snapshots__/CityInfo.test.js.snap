// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CityInfo snapshot renders error 1`] = `
<div
  className="city-info-error"
>
  <p>
     
    Location not found
     
  </p>
  <form
    onSubmit={[Function]}
  >
    <input
      type="submit"
      value="X"
    />
  </form>
</div>
`;

exports[`CityInfo snapshot renders info 1`] = `
<div
  className="city-info"
>
  <div
    className="city-info-head"
  >
    <p>
       
      London
       
    </p>
    <img
      src="http://openweathermap.org/img/wn/50n@2x.png"
    />
    <p>
       
      Â°C 
    </p>
    <form
      onSubmit={[Function]}
    >
      <input
        type="submit"
        value="X"
      />
    </form>
  </div>
  <div
    className="city-info-weather-list"
  >
    <div
      className="city-info-weather-list-item"
    >
      <p>
         
        Wind
         
      </p>
      <p>
         
        winddegdeg
        ; windspd m/s
         
      </p>
    </div>
    <div
      className="city-info-weather-list-item"
    >
      <p>
         
        Clouds
         
      </p>
      <p>
         
        desc
         
      </p>
    </div>
    <div
      className="city-info-weather-list-item"
    >
      <p>
         
        Pressure
         
      </p>
      <p>
         
        pressure hpa
         
      </p>
    </div>
    <div
      className="city-info-weather-list-item"
    >
      <p>
         
        Humidity
         
      </p>
      <p>
         
        humidity%
         
      </p>
    </div>
    <div
      className="city-info-weather-list-item"
    >
      <p>
         
        Coordinates
         
      </p>
      <p>
         
        [lat
        ; lon]
         
      </p>
    </div>
  </div>
</div>
`;

exports[`CityInfo snapshot renders loading 1`] = `
<div
  className="city-info-loading"
>
  <p>
     Loading, please wait... 
  </p>
  <img
    src="https://miro.medium.com/max/489/0*cWpsf9D3g346Va20.gif"
  />
</div>
`;
